core/tests/protractor_utils/AdminPage.js

```diff
@@ -292,6 +334,190 @@ var AdminPage = function() {
+    var text = 'No. of opportunities model created: 0';
+    await waitFor.visibilityOf(regenerationMessage,
+      'Regeneration message not visible');
+    expect(regenerationMessage.getText()).toEqual(text);
+  };
+
+  this.changeUsername = async function(oldUsername, newUsername) {
+    await waitFor.visibilityOf(oldUsernameInput,
+      'Current username input not visible');
+    await oldUsernameInput.sendKeys(oldUsername);
+    await waitFor.visibilityOf(newUsernameInput,
+      'New username input not visible');
+    await newUsernameInput.sendKeys(newUsername);
+    await waitFor.visibilityOf(usernameChangeSubmitButton,
+      'Change username submit button not visible');
+    await waitFor.elementToBeClickable(usernameChangeSubmitButton,
+      'Username change submit button not clickable');
+    await usernameChangeSubmitButton.click();
+    var text = 'Successfully renamed ' + oldUsername + ' to ' + newUsername;
+    await waitFor.textToBePresentInElement(statusMessage, text,
+      'Username was not successfully changed');
+  };
+
+  this.expectUsernameToBeChanged = async function(newUsername) {
+    expect(usernameSection.getText()).toEqual(newUsername);
+  };
+
+  this.fetchSVG = async function() {
+    await action.click('Start SVG Button', startSVGButton);
+  };
+
+  this.expectSVGToBeFetched = async function() {
+    await waitFor.visibilityOf(
+      statusMessage, 'Status message not visible');
+    await waitFor.textToBePresentInElement(
+      statusMessage, 'SVGs generated for all explorations .');
+  };
+
+  this.saveSVG = async function() {
+    await action.click('Save SVG Button', stopSVGButton);
+  };
+
+  this.expectSVGToBeSaved = async function() {
+    await waitFor.visibilityOf(
+      statusMessage, 'Status message not visible');
+    await waitFor.textToBePresentInElement(
+      statusMessage, 'Successfully updated 0 explorations, 0 left');
+  };
+
+  this.extractDataAndExpectExtraction = async function(expID, expVersion,
+      stateName, numberOfAnswers) {
+    await browser.refresh();
+    await waitFor.pageToFullyLoad();
+    await action.sendKeys('Extract Data Exploration ID Input',
+      extractDataExplorationIdInput, expID);
+    await action.sendKeys('Extract Data Exploration Version Input',
+      extractDataExplorationVersionInput, expVersion);
+    await action.sendKeys('Extract Data State Name Input',
+      extractDataStateNameInput, stateName);
+    await action.sendKeys('Extract Data Number Of Answers Input',
+      extractDataNumAnswersInput, numberOfAnswers);
+    await action.click('Extract Data Form Submit Button',
+      extractDataFormSubmitButton);
+    var handles = await browser.driver.getAllWindowHandles();
+    await action.click('Extract Data Submit Button',
+      extractDataFormSubmitButton);
+    var newHandles = await browser.driver.getAllWindowHandles();
+    expect(newHandles.length).toEqual(handles.length + 1);
```

Comment:
Can you close the newly opened window? Otherwise it might mess up other tests. In general we don't require that tests clean up after themselves, but I feel like we should for opening windows

Reply:
Done!